### PR TITLE
Update contentFor tag grammar to properly extract arguments

### PR DESCRIPTION
**Pull request by veken1199**
## What are you adding in this PR?
It seems that the grammar we had for content_for did not properly extract the contentFor liquid tag argument and convert them into array. 

In this PR we have:
1. Updated the grammar for contentFor liquid tag to support arguments that contain `.`
2. Updated ValidContentForArguments check to support `closest.` instead of `context.`
3. Introduced a new `DeprecatedContentForArgument` check that reports a warning when contentFor tag contains `context.`

- [ ] This PR includes a new checks or changes the configuration of a check
  - [x] I included a minor bump `changeset`
  - [ ] It's in the `allChecks` array in `src/checks/index.ts`
  - [ ] I ran `yarn build` and committed the updated configuration files
    
    
    
    - [ ] If applicable, I've updated the `theme-app-extension.yml` config

- [ ] I included a minor bump `changeset`
- [x] My feature is backward compatible

- [ ] I included a patch bump `changeset`

### CODE REVIEW

File: packages/liquid-html-parser/src/stage-1-cst.ts
Comment (charlespwd) [inline packages/liquid-html-parser/src/stage-1-cst.ts:385]: Does this really need a new concrete node type? I see we fold it back into NamedArgument in stage 2. Why not fold it in the contentForTagArgument mapping instead?

File: packages/liquid-html-parser/grammar/liquid-html.ohm
Comment (charlespwd) [inline packages/liquid-html-parser/grammar/liquid-html.ohm:135]: Is this because the `.` is only accepted in content_for named argument parsing?

Comment (veken1199) [reply packages/liquid-html-parser/grammar/liquid-html.ohm:135]: Yes indeed. I don't believe other tag attributes need to support `.` atm. This is why I had to create new contentForTagArgument

File: packages/theme-check-common/src/checks/deprecated-content-for-arguments/index.ts
Comment (charlespwd) [inline packages/theme-check-common/src/checks/deprecated-content-for-arguments/index.ts:71]: Given how transient this thing is and the thing hasn't been released. We can probably not ship this. Just let the one that accepts `closest` complain about `context` not being supported.

Comment (charlespwd) [reply packages/theme-check-common/src/checks/deprecated-content-for-arguments/index.ts:71]: Thing is this could live in this repo forever. People will start being able to have that in their `.theme-check.yml` and I can't remember what the behaviour is for when you have settings for checks that don't exist. 

Which would probably mean that removing the check would require a major version bump of theme-check. 

Do we have another check that we could use instead? Something that wouldn't go away? Or maybe something like DeprecatedFeature?

Comment (charlespwd) [reply packages/theme-check-common/src/checks/deprecated-content-for-arguments/index.ts:71]: That's probably a nice place for it yeah!

Comment (veken1199) [reply packages/theme-check-common/src/checks/deprecated-content-for-arguments/index.ts]: dropped this check

**Review by albchu — APPROVED** [review]
LGTM!

File: packages/liquid-html-parser/grammar/liquid-html.ohm
Comment (charlespwd) [inline packages/liquid-html-parser/grammar/liquid-html.ohm:135]: Can we de-indent those two lines? They are new rules and should be at the same indentation level as the other rule declarations. 
```suggestion
  contentForTagArgument = listOf<contentForNamedArgument<delimTag>, argumentSeparatorOptionalComma>
  contentForNamedArgument<delim> = (variableSegment ("." variableSegment)*) space* ":" space* (liquidExpression<delim>)
```

**Review by charlespwd — APPROVED** [review]
Other than the nitpick, this LGTM :) Feel free to merge after!

**Pull request merged**
